File: Leetcode/power_of_two/power_of_two.py
def is_power_of_two(n):
    x = 0
    
    def recursion(n):
        nonlocal x 
        if 2 ** x < n:
            x += 1
            return recursion(n)
        elif 2 ** x == n:
            return True
        else:
            return False

    return recursion(n)


print(is_power_of_two(1))  
print(is_power_of_two(16))  
print(is_power_of_two(3))

File: Leetcode/power_of_two/power_of_two.rb
def is_power_of_two(n)
  @x = 0
  def recursion(n)
    if 2 ** @x < n
      @x += 1
      recursion(n)
    elsif 2 ** @x == n
      return true
    else 2 ** @x > n
      return false
    end
  end
  recursion(n)
end

p is_power_of_two(1)
p is_power_of_two(16)
p is_power_of_two(3)
